### ISSUE TITLE
 'NoneType' object has no attribute 'log'

### ISSUE THREAD

**Issue by urograndis** [semiautomaticgit/SemiAutomaticClassificationPlugin#362]
Good morning.
I am running QGIS 3.38.0-Grenoble on a 64-bit platform, Ubuntu 22.04.4 LTS.
I had the Semi-Automatic Classification Plugin installed and was running the Region Growing Algorithm when it crashed and closed.
When I restarted, I got the message: Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'.
I uninstalled and reinstalled the plugin: Plugin installed successfully;
But it doesn't even appear in the taskbar and the message:
2024-07-30T09:29:41 WARNING Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'
2024-07-30T09:29:42 INFO Plugin installed successfully.
So, I need your help.

**Comment by semiautomaticgit** [comment]
Hello @urograndis ,
it seems an installation issue.
Please update the plugin and Remotior Sensus following these installation steps https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation.html

**Comment by semiautomaticgit** [comment]
Hello @urograndis ,
please update the SCP to version 8.3 and report if you still get the same error

**Comment by kleinhoellental** [comment]
Hello Luca, 
unfortunately i got the same issue using: 

- QGIS 3.38.0
- macOS Sonoma 14.6.1
- SCP 8.3
- remotior-sensus 0.4

works partially on the QGIS-LTR 

- Classification Preview works
- Classification itself doesn't

regards
Seb

**Comment by Leandro-G** [comment]
Hello Luca. Same issue here. No chance to enable "Records detailed events in a log file" in the plugin due to it failing to load. Running QGIS 3.38 in Ubuntu 24.04 and SCP 8.3.
I get this error: "Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'"
Thank you in advance. Your plugin is great for teaching purposes.

**Comment by semiautomaticgit** [comment]
> Hello Luca. Same issue here. No chance to enable "Records detailed events in a log file" in the plugin due to it failing to load. Running QGIS 3.38 in Ubuntu 24.04 and SCP 8.3. I get this error: "Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'" Thank you in advance. Your plugin is great for teaching purposes.

Hello @Leandro-G , this error is related to installation.
I suggest to uninstall QGIS and the plugin, then reinstall it again following this guide https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_debian.html

**Comment by Leandro-G** [comment]
> > Hello Luca. Same issue here. No chance to enable "Records detailed events in a log file" in the plugin due to it failing to load. Running QGIS 3.38 in Ubuntu 24.04 and SCP 8.3. I get this error: "Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'" Thank you in advance. Your plugin is great for teaching purposes.
> 
> Hello @Leandro-G , this error is related to installation. I suggest to uninstall QGIS and the plugin, then reinstall it again following this guide https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_debian.html

Thank you Luca for your answer. Followed exactly, no error messages during installation but I get the exact same message and SCP does not load...

**Comment by semiautomaticgit** [comment]
> > > Hello Luca. Same issue here. No chance to enable "Records detailed events in a log file" in the plugin due to it failing to load. Running QGIS 3.38 in Ubuntu 24.04 and SCP 8.3. I get this error: "Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'" Thank you in advance. Your plugin is great for teaching purposes.
> > 
> > 
> > Hello @Leandro-G , this error is related to installation. I suggest to uninstall QGIS and the plugin, then reinstall it again following this guide https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_debian.html
> 
> Thank you Luca for your answer. Followed exactly, no error messages during installation but I get the exact same message and SCP does not load...

Thank you for your feedback. 
Unfortunately I can't reproduce the issue. I think it is related to the system, or the QGIS Python environment.
I suggest to perform a clean installation for example with a new user, or installing through Conda as described here https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_conda.html

**Comment by Leandro-G** [comment]
> > > > Hello Luca. Same issue here. No chance to enable "Records detailed events in a log file" in the plugin due to it failing to load. Running QGIS 3.38 in Ubuntu 24.04 and SCP 8.3. I get this error: "Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'" Thank you in advance. Your plugin is great for teaching purposes.
> > > 
> > > 
> > > Hello @Leandro-G , this error is related to installation. I suggest to uninstall QGIS and the plugin, then reinstall it again following this guide https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_debian.html
> > 
> > 
> > Thank you Luca for your answer. Followed exactly, no error messages during installation but I get the exact same message and SCP does not load...
> 
> Thank you for your feedback. Unfortunately I can't reproduce the issue. I think it is related to the system, or the QGIS Python environment. I suggest to perform a clean installation for example with a new user, or installing through Conda as described here https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_conda.html

You're right Luca. It works in Debian 12, tried on a different machine. It's an Ubuntu isuue

**Comment by semiautomaticgit** [comment]
> > > > > Hello Luca. Same issue here. No chance to enable "Records detailed events in a log file" in the plugin due to it failing to load. Running QGIS 3.38 in Ubuntu 24.04 and SCP 8.3. I get this error: "Semi-Automatic Classification Plugin: 'NoneType' object has no attribute 'log'" Thank you in advance. Your plugin is great for teaching purposes.
> > > > 
> > > > 
> > > > Hello @Leandro-G , this error is related to installation. I suggest to uninstall QGIS and the plugin, then reinstall it again following this guide https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_debian.html
> > > 
> > > 
> > > Thank you Luca for your answer. Followed exactly, no error messages during installation but I get the exact same message and SCP does not load...
> > 
> > 
> > Thank you for your feedback. Unfortunately I can't reproduce the issue. I think it is related to the system, or the QGIS Python environment. I suggest to perform a clean installation for example with a new user, or installing through Conda as described here https://semiautomaticclassificationmanual.readthedocs.io/en/latest/installation_conda.html
> 
> You're right Luca. It works in Debian 12, tried on a different machine. It's an Ubuntu isuue

Thank you very much for your feedback. I'll try to understand if there is something I can do

**Comment by semiautomaticgit** [comment]
Hello @urograndis @Leandro-G @kleinhoellental ,

you may try this alternative method of installation of dependencies https://semiautomaticclassificationmanual.readthedocs.io/en/latest/faq.html#plugin-installation-4
